frontend/src/models/User.ts

```diff
@@ -1,25 +1,19 @@
-import { IAuthResponse } from "@/models/IAuthResponse";
-
-export default class User {
-  userName: string = "";
-  name: string = "Login / Sign Up";
-  loggedIn: boolean = false;
-  profileImage: string = "";
-
-  constructor(response?: IAuthResponse) {
-    if (response) {
-      this.loggedIn = true;
-      this.userName = response.user_id;
-
-      for (let claim of response.user_claims) {
-        if (claim.typ.indexOf("/identity/claims/name") > 0) {
-          this.name = claim.val;
-        }
-
-        if (claim.typ.indexOf("profile_image_url_https") > 0) {
-          this.profileImage = claim.val;
-        }
-      }
-    }
-  }
-}
+import jsonwebtoken from "jsonwebtoken";
+
+export default class User {
+  userName: string = "";
+  name: string = "Login / Sign Up";
+  loggedIn: boolean = false;
+  profileImage: string = "";
+
+  constructor(jwtToken?: string) {
+    if (jwtToken) {
+      const claims: any = jsonwebtoken.decode(jwtToken);
+
+      this.name = `${claims.given_name} ${claims.family_name}`;
```

Comment:
Should we add any validation here around the different claim attributes? I would assume most things will have a `name` and `email`, but I have seen in weird situations some AAD principals don't have `emails`. (Ex: service principals)

Reply:
AD B2C seems to be normalizing them all to these values for the currently configured providers.  We can loop back if we start seeing inconsistency.